src/bentoml/_internal/marshal/dispatcher.py

```diff
@@ -401,3 +401,46 @@ async def outbound_call(self, inputs_info: tuple[Job, ...]):
                     if not fut.done():
                         fut.cancel()
             self._sema.release()
+
+
+class NoOpDispatcher:
```

Comment:
I don't know if this was addressed already, but was there a reason you couldn't use the normal dispatcher with batching disabled?

Reply:
It was mainly for convenience without conflating the current implementation, I could give it a try with the `CorkDispatcher`

Reply:
The biggest difference here is that for streaming, the `callback` returns a async generator, and it is handled differently for a normal response. in particular this part of the code.
```python
            outputs = await self.callback(
                tuple(t.cast(t.Any, input_info.data) for input_info in inputs_info)
            )
            assert len(outputs) == len(inputs_info)
            for input_info, out in zip(inputs_info, outputs):
                fut = input_info.future
                if not fut.done():
                    fut.set_result(out)
            _done = True
            self.optimizer.log_outbound(
                n=len(inputs_info),
                wait=_time_start - inputs_info[-1].enqueue_time,
                duration=time.time() - _time_start,
            )
```

Not sure if it's worth conflating the implementation here to use a single dispatcher?

Reply:
And speaking of this, I think we might need to give it a better name? Since its also a dispatcher to handle generators?

Reply:
I think we should update the old dispatcher to support streaming in every case, no real reason not to.

Reply:
Just trying to understand, what are the benefits by using the current dispatcher?

Reply:
I believe that all the cancellation logic still applies in a streaming context, and besides, I think every response should be streamed anyway, which means that having two doesn't really make sense.

Reply:
Yep!